frontend/app/helpers/spree/addresses_helper.rb

```diff
@@ -0,0 +1,36 @@
+# https://github.com/spree-contrib/spree_address_book/blob/master/app/helpers/spree/addresses_helper.rb
+module Spree
+  module AddressesHelper
+    def address_field(form, method, address_id = 'b', &handler)
+      content_tag :p, id: [address_id, method].join, class: 'form-group' do
+        if handler
+          yield
+        else
+          is_required = Spree::Address.required_fields.include?(method)
+          separator = is_required ? '<span class="required">*</span><br />' : '<br />'
+          form.label(method) + separator.html_safe +
+            form.text_field(method, class: [is_required ? 'required' : nil, 'form-control'].compact, required: is_required)
+        end
+      end
+    end
+
+    def address_state(form, country, _address_id = 'b')
+      country ||= Spree::Country.find(Spree::Config[:default_country_id])
+      have_states = country.states.any?
+      state_elements = [
```

Comment:
Please move 'state_elements' and 'form.label(:state, Spree.t(:state))...' to a separate method.